### PR TITLE
WIP: Rewrite of the operator with Unstructured

### PR DESCRIPTION
**Pull request by zvonkok**
- Bump up SDK version
- Switched to Unstructured API

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,70 +1,63 @@
 REGISTRY       ?= quay.io
-ORG            ?= zvonkok
-TAG            ?= latest #$(shell git rev-parse --short HEAD)
+ORG            ?= openshift-psap
+TAG            ?= $(shell git branch | grep \* | cut -d ' ' -f2)
```
Comment (jmencak) [inline Makefile]: Are you trying to get current branch here?  Won't `git rev-parse --abbrev-ref HEAD` work?

File: assets/state-worker.yaml [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+    sources:
+      pci:
+        deviceLabelFields:
+          - "vendor"
+        deviceClassWhitelist:
+          - "0200"
+          - "03"
+          - "12"
+    
+    #      - "class"
```
Comment (jmencak) [inline assets/state-worker.yaml]: Why is this and the following 3 lines commented out?

File: go.mod [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+	k8s.io/api v0.0.0-20190612125737-db0771252981
+	k8s.io/apimachinery v0.0.0-20190612125636-6a5db36e93ad
+	k8s.io/client-go v11.0.0+incompatible
+	k8s.io/kube-openapi v0.0.0-20190603182131-db7b694dc208
+	sigs.k8s.io/controller-runtime v0.1.12
+	sigs.k8s.io/controller-tools v0.1.10
+	sigs.k8s.io/yaml v1.1.0
+)
+
+// Pinned to kubernetes-1.13.4
```
Comment (jmencak) [inline go.mod]: Shouldn't we be writing for k8s 1.16 now?

File: pkg/controller/nodefeaturediscovery/waitfor.go [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+
+// WAIT FOR RESOURCES -- other file?
+
+var (
+	retryInterval = time.Second * 5
+	timeout       = time.Second * 60
+)
+
+func waitForResourceAvailability(obj *unstructured.Unstructured, r *ReconcileNodeFeatureDiscovery) error {
+
```
Comment (jmencak) [inline pkg/controller/nodefeaturediscovery/waitfor.go]: Is the new line here intentional?

File: pkg/controller/nodefeaturediscovery/waitfor.go [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+			}
+			return false, err
+		}
+		return true, nil
+	})
+	return err
+}
+
+func waitForResourceFullAvailability(obj *unstructured.Unstructured, r *ReconcileNodeFeatureDiscovery, callback statusCallback) error {
+
```
Comment (jmencak) [inline pkg/controller/nodefeaturediscovery/waitfor.go]: Is the new line here intentional?

File: Makefile (hunk 1 above)
Comment (zvonkok) [reply Makefile]: You're right, done.

File: assets/state-worker.yaml (hunk 2 above)
Comment (zvonkok) [reply assets/state-worker.yaml]: This is the config file, left it as a hint how the config file looks like.

File: go.mod (hunk 3 above)
Comment (zvonkok) [reply go.mod]: Yep need to update the module. Thanks for the pointer.

Comment (zvonkok) [reply go.mod]: Hmm,  thinking about it, do we really need the latest kubernetes module? I am not using any features that are not in 1.13 ...

File: pkg/controller/nodefeaturediscovery/waitfor.go (hunk 4 above)
Comment (zvonkok) [reply pkg/controller/nodefeaturediscovery/waitfor.go]: Removed.

File: pkg/controller/nodefeaturediscovery/waitfor.go (hunk 5 above)
Comment (zvonkok) [reply pkg/controller/nodefeaturediscovery/waitfor.go]: Removed.

File: go.mod (hunk 3 above)
Comment (jmencak) [reply go.mod]: Far from being an expert on this, but I try to match this [compatilibity matrix](https://github.com/kubernetes/client-go#compatibility-client-go---kubernetes-clusters) at least for client-go.  Waiting a while for the table update for 1.16 for my operator.

Comment (zvonkok) [reply go.mod]: Ah good pointer bumped the client-go to release-13.0

**Pull request closed without merging**